### PR TITLE
Add support for view comments for Snowflake

### PR DESCRIPTION
**Pull request by bombsimon**
Snowflake supports adding a comment when creating a view, see https://docs.snowflake.com/en/sql-reference/sql/create-view#syntax.

### CODE REVIEW

File: src/parser/mod.rs [hunk 1]
```diff
@@ -3903,6 +3903,21 @@ impl<'a> Parser<'a> {
             };
         }
 
+        let comment = if dialect_of!(self is SnowflakeDialect) {
+            if self.parse_keyword(Keyword::COMMENT) {
+                let _ = self.consume_token(&Token::Eq);
+                let next_token = self.next_token();
+                match next_token.token {
+                    Token::SingleQuotedString(str) => Some(str),
+                    _ => self.expected("comment", next_token)?,
```
Comment (iffyio) [inline src/parser/mod.rs:3912]:
```suggestion
                    _ => self.expected("string literal", next_token)?,
```

File: tests/sqlparser_common.rs [hunk 2]
```diff
@@ -6387,6 +6394,46 @@ fn parse_create_or_replace_view() {
...
+            assert!(comment.is_none());
+            assert!(!late_binding);
+            assert!(!if_not_exists);
+            assert!(!temporary);
+        }
+        _ => unreachable!(),
+    }
+}
+
+#[test]
```
Comment (iffyio) [inline tests/sqlparser_common.rs:6406]: we can put the test in [sqlparser_snowflake.rs](https://github.com/sqlparser-rs/sqlparser-rs/blob/33963675d75c11e64e423fd79ab998193dbb4286/tests/sqlparser_snowflake.rs)

File: src/parser/mod.rs [hunk 3]
```diff
@@ -3903,6 +3903,21 @@ impl<'a> Parser<'a> {
             };
         }
 
+        let comment = if dialect_of!(self is SnowflakeDialect) {
```
[the remark this answers is not in the document]
Comment (bombsimon) [reply src/parser/mod.rs]: Oops, missed this one. Thanks for linking, updated!

File: src/ast/mod.rs [hunk 4]
```diff
@@ -3239,6 +3243,9 @@ impl fmt::Display for Statement {
                     temporary = if *temporary { "TEMPORARY " } else { "" },
                     if_not_exists = if *if_not_exists { "IF NOT EXISTS " } else { "" }
                 )?;
+                if let Some(comment) = comment {
+                    write!(f, " COMMENT = '{comment}'")?;
```
Comment (jmhain) [inline src/ast/mod.rs:3247]:
```suggestion
                    write!(f, " COMMENT = '{}'", value::escape_single_quote_string(comment))?;
```

**Review by jmhain — APPROVED** [review]
lgtm, modulo the missing string escape cc @alamb

**Review by iffyio — APPROVED** [review]

**Review by alamb — APPROVED** [review]
Thank you for the contributon @bombsimon  and thank you for the reviews @iffyio  and @jmhain

**Pull request merged**